internal/pkg/repository/repository.go

```diff
@@ -0,0 +1,77 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// package repository provides support for building and pushing images to a repository.
+package repository
+
+import (
+    "fmt"
+    "github.com/aws/copilot-cli/internal/pkg/aws/ecr"
+)
+
+// DockerService provides support for logging in to repositories, building images and pushing images to repositories.
+type DockerService interface {
```

Comment:
What do you think of:
```suggestion
type ContainerManager interface {
```